xrtpy/util/xrt_remove_lightleak.py

```diff
@@ -0,0 +1,275 @@
+    rupture development occurred multiple times as follows:
+
+    phase 1 :  9-May-2012 12:00
+    phase 2 : 14-Jun-2015 12:30
+    phase 3 : 27-May-2017 11:00
+    phase 4 : 29-May-2018 00:00
+    phase 5 :  8-Jun-2022 12:40
+
+    2. The light leak correction is done by simply subtracting the light leak
+    image (visible stray light component included in each X-ray filter pair)
+    obtained during the Hinode satellite's eclipse season, which occurs
+    roughly May to August each year.  The light leak image has the following
+    characteristics:
+
+    * Leak pattern and intensity differ with each filter and also with the
+      satellite pointing.
+    * Leak pattern and intensity of the same filter and pointing are roughly
+      constant during each stray light phase (above), but vary by ~10%
+      depending on the growth of the contamination layer on the CCD (that
+      repeats gradual increase and jump down after CCD bakeouts).
+
+    3. The image for the light leak correction should therefore be selected
+    for the same filter, same satellite pointing, same stray light phase then
+    ideally be adjusted for the level of CCD contamination at the time of
+    observation.  However, as a practical matter, preparing the leak image for
+    every possible pointing is hard to achieve, while we have good amount of
+    light leak measurements at the disk center pointing. The full-disk
+    composite images are therefore corrected most reliably for the light leak.
+
+    4. Intensity variation of the leak image due to the growth of CCD
+    contamination layer is well determined for Ti_poly at the stray light
+    phase 1 by using the intensity correlation between Ti_poly and Al_mesh
+    images (cg. :cite:p:`Takeda:2016`, p.317). The resulting
+    k-factor is obtained with the function :file:`GET_SLCORFACT_RAW.PRO`, and the leak
+    image subtraction has been already performed only for the Ti_poly SCIA
+    images at the phase 1 (as of Feb-2022).
+
+    """
+
+    # Check to see if the lightleak has already been subtracted
+    history = in_map.meta["HISTORY"]
+    if "Light leak subtraction: DONE" in history:
+        raise ValueError(
+            "HISTORY indicates light leak subtraction already done on image."
+        )
+    # ********* select leak image from the archive *********
+    if leak_image is None:
+        dir_leak = Path(__file__).parent.absolute() / "data" / "leak_fits"
+        filt1 = in_map.meta["EC_FW1"]
+        filt2 = in_map.meta["EC_FW2"]
+        fpair = str(filt1) + str(filt2)
+        sfilt1 = in_map.meta["EC_FW1_"]
+        sfilt2 = in_map.meta["EC_FW2_"]
+        sfpair = f"{sfilt1}/{sfilt2}"
+        sl_phase = check_sl_phase(in_map.meta["date_obs"])
+        if sl_phase == 6:
+            print(
+                "Warning: light leak images for this period are not yet"
+                " available. Defaulting to previous phase."
+            )
```

Comment:
```suggestion
            warnings.warn(
                "light leak images for this period are not yet"
                " available. Defaulting to previous phase."
            )
```
...would also need to `import warnings` above